src/js/tabs/firmware_flasher.js

```diff
@@ -923,7 +926,10 @@ firmware_flasher.initialize = function (callback) {
                 tracking.sendEvent(tracking.EVENT_CATEGORIES.FLASHING, 'ExitDfu', null);
                 try {
                     console.log('Closing DFU');
-                    DFU.connect(self.parsed_hex, { exitDfu: true });
+                    DFU.requestPermission()
+                    .then((device) => {
+                        DFU.connect(device.path, self.parsed_hex, { exitDfu: true });
+                    });
```

Comment:
I had designed the USB (DFU) API to prevent duplication.
Maybe we have another method in the API which does this ~~like before~~ with a single command.

Perhaps
```bash
DFU.requestPermissionAndConnect(path, hex, options, callback);
```
```

Reply:
The idea is to remove this and work with events. This is only an intermediate pass before the final code. 
I thought that the objective is to have automatic fishing without asking permission if possible.